sdk/communication/AzureCommunicationChat/Source/Signaling/CommunicationSignalingClient.swift

```diff
@@ -98,18 +135,73 @@ class CommunicationSignalingClient {
     }
 
     func on(event: ChatEventId, handler: @escaping TrouterEventHandler) {
+        // Guard to check if selfHostedTrouterClient is initialized
+        guard let selfHostedTrouterClient = self.selfHostedTrouterClient else {
+            logger.error("Failed to register event handler: SelfHostedTrouterClient is not initialized.")
+            return
+        }
+        
         let logger = ClientLoggers.default(tag: "AzureCommunicationHandler-\(event)")
         let communicationHandler = CommunicationHandler(handler: handler, logger: logger)
         selfHostedTrouterClient.register(communicationHandler, forPath: "/\(event)")
         communicationHandlers[event] = communicationHandler
     }
 
     func off(event: ChatEventId) {
+        // Guard to check if selfHostedTrouterClient is initialized
+        guard let selfHostedTrouterClient = self.selfHostedTrouterClient else {
+            logger.error("Failed to unregister event handler: SelfHostedTrouterClient is not initialized.")
+            return
+        }
+        
         if let communicationHandler = communicationHandlers[event] {
             selfHostedTrouterClient.unregisterListener(communicationHandler)
             communicationHandlers.removeValue(forKey: event)
         }
     }
+    
+    private func getTrouterSettings(
+            token: String,
+            endpoint: String,
+            completionHandler: @escaping (Result<RealTimeNotificationConfiguration, AzureError>) -> Void
+        ) {
+            let urlString = "\(endpoint)/chat/config/realTimeNotifications?api-version=2024-09-01"
```

Comment:
Please put the api version into a separate constant

Reply:
Put it in the constant "configApiVersion" right now.